src/libraries/System.Runtime.Loader/tests/ApplyUpdateTest.cs

```diff
@@ -179,6 +180,41 @@ public void CustomAttributeDelete()
             });
         }
 
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/52993", TestRuntimes.Mono)]
+        [ConditionalFact(typeof(ApplyUpdateUtil), nameof (ApplyUpdateUtil.IsSupported))]
+        public void AsyncMethodChanges()
+        {
+            // Test that changing an async method doesn't cause any type load exceptions
+            ApplyUpdateUtil.TestCase(static () =>
+            {
+                Assembly assembly = typeof(System.Reflection.Metadata.ApplyUpdate.Test.AsyncMethodChange).Assembly;
+
+                ApplyUpdateUtil.ApplyUpdate(assembly);
+                ApplyUpdateUtil.ClearAllReflectionCaches();
+
+                Type ty = typeof(System.Reflection.Metadata.ApplyUpdate.Test.AsyncMethodChange);
+                Assert.NotNull(ty);
+
+                MethodInfo mi = ty.GetMethod(nameof(System.Reflection.Metadata.ApplyUpdate.Test.AsyncMethodChange.TestTaskMethod), BindingFlags.Public | BindingFlags.Static);
+                Assert.NotNull(mi);
+
+                string result = ApplyUpdate.Test.AsyncMethodChange.TestTaskMethod().GetAwaiter().GetResult();
+                Assert.Equal("TestTaskMethod v1", result);
+
+                object[] attributes = mi.GetCustomAttributes(true);
+                Assert.NotNull(attributes);
+                Assert.True(attributes.Length > 0);
+
+                foreach (var attribute in attributes)
+                {
+                    if (attribute is AsyncStateMachineAttribute asm)
+                    {
+                        Console.WriteLine(asm.StateMachineType);
```

Comment:
Should this assert something instead?

Reply:
Oops. Yes, that should assert something.